source/client/options_impl.cc

```diff
@@ -121,40 +161,153 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
     throw NoServingException();
   }
 
-  requests_per_second_ = requests_per_second.getValue();
-  connections_ = connections.getValue();
-  duration_ = duration.getValue();
-  timeout_ = timeout.getValue();
+  if (requests_per_second.isSet()) {
+    requests_per_second_ = requests_per_second.getValue();
+  }
+  if (connections.isSet()) {
+    connections_ = connections.getValue();
+  }
+  if (duration.isSet()) {
+    duration_ = duration.getValue();
+  }
+  if (timeout.isSet()) {
+    timeout_ = timeout.getValue();
+  }
   uri_ = uri.getValue();
-  h2_ = h2.getValue();
-  concurrency_ = concurrency.getValue();
-  verbosity_ = verbosity.getValue();
-  output_format_ = output_format.getValue();
-  prefetch_connections_ = prefetch_connections.getValue();
-  burst_size_ = burst_size.getValue();
-  address_family_ = address_family.getValue();
-  request_method_ = request_method.getValue();
-  request_headers_ = request_headers.getValue();
-  request_body_size_ = request_body_size.getValue();
-
-  // We cap on negative values. TCLAP accepts negative values which we will get here as very
-  // large values. We just cap values to 2^63.
-  const uint64_t largest_acceptable_uint64_option_value =
-      static_cast<uint64_t>(std::pow(2ull, 63ull));
-
-  if (requests_per_second_ == 0 || requests_per_second_ > largest_acceptable_uint64_option_value) {
+  if (h2.isSet()) {
+    h2_ = h2.getValue();
+  }
+  if (concurrency.isSet()) {
+    concurrency_ = concurrency.getValue();
+  }
+  if (verbosity.isSet()) {
+    verbosity_ = verbosity.getValue();
+  }
+  if (output_format.isSet()) {
+    output_format_ = output_format.getValue();
+  }
+  if (prefetch_connections.isSet()) {
+    prefetch_connections_ = prefetch_connections.getValue();
+  }
+  if (burst_size.isSet()) {
+    burst_size_ = burst_size.getValue();
+  }
+  if (address_family.isSet()) {
+    address_family_ = address_family.getValue();
+  }
+  if (request_method.isSet()) {
+    request_method_ = request_method.getValue();
+  }
+  if (request_headers.isSet()) {
+    request_headers_ = request_headers.getValue();
+  }
+  if (request_body_size.isSet()) {
+    request_body_size_ = request_body_size.getValue();
+  }
+  if (max_pending_requests.isSet()) {
+    max_pending_requests_ = max_pending_requests.getValue();
+  }
+  if (max_active_requests.isSet()) {
+    max_active_requests_ = max_active_requests.getValue();
+  }
+  if (max_requests_per_connection.isSet()) {
+    max_requests_per_connection_ = max_requests_per_connection.getValue();
+  }
+
+  // CLI-specific tests.
+  if (requests_per_second_ > largest_acceptable_uint32_option_value) {
```

Comment:
Might be nice to accumulate these errors so the user can see everything they did wrong all at once.